samples/emoji-search/presenter/src/commonTest/kotlin/com/example/redwood/emojisearch/presenter/MutableWidget.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (C) 2023 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.redwood.emojisearch.presenter
+
+import app.cash.redwood.RedwoodCodegenApi
+import app.cash.redwood.widget.Widget
+
+@RedwoodCodegenApi
+interface MutableWidget : Widget<MutableWidget> {
```

Comment:
I ‘generated’ two classes for each widget, like `MutableTextInput` and `TextInputValue`. The mutable one is implementation-detail and hidden from the public API with `@RedwoodCodegenApi`. The value one is almost a data class, but excluding lambas from equals/hashCode/toString. This whole dual hierarchy is necessary 'cause Kotlin doesn’t have lenses.